src/nvim/testdir/test_functions.vim

```diff
@@ -818,6 +864,19 @@ func Test_filewritable()
   bw!
 endfunc
 
+func Test_Executable()
+  if has('win32')
+    call assert_equal(1, executable('notepad'))
+    call assert_equal(1, executable('notepad.exe'))
+    call assert_equal(0, executable('notepad.exe.exe'))
+    call assert_equal(0, executable('shell32.dll'))
+    call assert_equal(0, executable('win.ini'))
```

Comment:
Why are these not executable? `dll,ini` don't open notepad?

```
Found errors in Test_Executable():
function RunTheTest[35]..Test_Executable line 5: Expected 0 but got 1
function RunTheTest[35]..Test_Executable line 6: Expected 0 but got 1
```

Reply:
Forgot that `executable` checks files in `PATH`. shell32.dll is in `PATH` but `dll` is not executable (won't open notepad as fallback).  `win.ini` is in `PATH` (C:\Windows\win.ini on Windows 8.1) and opens notepad so it is executable. notepad.exe.exe is not in `PATH` (unless it was created elsewhere) so it is not executable here.

Reply:
`win.ini` is loaded to notepad because of the default file assocation between `inifile` and `notepad` on Windows.
In cmd.exe, user can associate file extensions with a filetype and a default command via [assoc](https://ss64.com/nt/assoc.html) and [ftype](https://ss64.com/nt/ftype.html) commands.